### PR TITLE
hd-wallet  - cleanup docs to match wallet-sdk

### PR DESCRIPTION
**Pull request by nillo**
added updated docs for hd-wallet to align with wallet-sdk

### CODE REVIEW

**Review by Randynamic — APPROVED** [review]

**Comment by Randynamic** [comment]
@lsgunnlsgunn can we check this and merge?

**Review by barthuijgen — APPROVED** [review]

File: packages/libs/hd-wallet/README.md [hunk 1]
```diff
@@ -51,128 +51,263 @@ The library is divided into two parts:
 
 Below are some common use cases for the `@kadena/hd-wallet` library.
 
-### Generating a Mnemonic
+## Key Concepts
 
-A mnemonic is a set of words that can be used to generate a seed. You can
-generate a mnemonic using `kadenaGenMnemonic`:
+- **Mnemonic Phrase**: A human-readable set of words used to generate a seed.
+- **Seed**: A binary representation of the mnemonic, which can be used to derive
```
Comment (javadkh2) [inline packages/libs/hd-wallet/README.md:57]: this is not correct! *Entropy* is the representation of the mnemonic not seed. 
Seed is a binary derived from mnemonic (with a hashing algorithm) will be used for creating master private key. Basically you can't generate mnemonic from a seed but you can create it from Entropy.

File: packages/libs/hd-wallet/README.md [hunk 2]
```diff
@@ -51,128 +51,263 @@ The library is divided into two parts:
...
-const txHash = 'tx-hash';
-const signer = kadenaSignWithKeyPair(password, publicKey, privateKey);
-const signature = await signer(txHash);
-console.log(signature); // Outputs the signature
+const isValid = kadenaVerify(txHash, publicKey, signature.sig);
+console.log(isValid); // Outputs true if the signature is valid
 ```
 
-### Retrieving Public Keys from Seed
+## Legacy Chainweaver Support
```
Comment (javadkh2) [inline packages/libs/hd-wallet/README.md:205]: ## Legacy Chainweaver (version 2) Support
```suggestion
## Legacy Chainweaver (version 2) Support
```

File: packages/libs/hd-wallet/README.md [hunk 3]
```diff
@@ -186,6 +321,55 @@ const decryptedSeed = kadenaDecrypt(password, seed);
...
+   storing.
+3. **Generate key pairs as needed** using indices (e.g., index 0 for the primary
+   account, index 1 for a secondary account).
+4. **Sign transactions** with the private key or directly with the seed.
+5. **Retrieve public keys** when you need to display addresses or verify
+   signatures.
+
+### Key Derivation Path
+
+The Kadena wallet uses the standard derivation path:
```
Comment (javadkh2) [inline packages/libs/hd-wallet/README.md:343]: add link to "packages/libs/hd-wallet/docs/decisions/0001-use-slip10-for-private-key-generation.md"

**Review by javadkh2 — CHANGES_REQUESTED** [review]

File: packages/libs/hd-wallet/README.md [hunk 4]
```diff
@@ -51,128 +51,263 @@ The library is divided into two parts:
...
+- **SLIP-0010**: The standard used for key derivation, ensuring compatibility
+  with other systems.
+
+## High-Level Steps
+
+1. **Generate a Mnemonic Phrase**
+2. **Convert Mnemonic to Seed**
+3. **Generate Key Pairs from Seed**
+4. **Sign Transactions with Keys**
+
```
[the remark this answers is not in the document]
Comment (barthuijgen) [reply packages/libs/hd-wallet/README.md:70]: This does not seem directly related to hd-wallet, this should be documented with `@kadena/client` itself. or if using sdk methods in the `@kadena/wallet-sdk` docs.

**Review by barthuijgen — APPROVED** [review]

**Review by javadkh2 — APPROVED** [review]

**Pull request merged**